mmcls/models/backbones/pyramid_vig.py

```diff
@@ -0,0 +1,162 @@
+        )
+
+    def forward(self, x):
+        x = self.conv(x)
+        return x
+
+
+@MODELS.register_module()
+class pyramid_vig(BaseBackbone):
+    # blocks, channels
+    arch_settings = {
+        'ti': [[2, 2, 6, 2], [48, 96, 240, 384]],
+        's': [[2, 2, 6, 2], [80, 160, 400, 640]],
+        'm': [[2, 2, 16, 2], [96, 192, 384, 768]],
+        'b': [[2, 2, 18, 2], [128, 256, 512, 1024]]
+    }
+
+    def __init__(self,
+                 model_cnf,
+                 k,
+                 act,
+                 norm,
+                 bias,
+                 epsilon,
+                 use_stochastic,
+                 conv,
+                 drop_path,
+                 dropout,
+                 n_classes,
+                 norm_eval=False,
+                 frozen_stages=0):
+        super(pyramid_vig, self).__init__()
+        model_cnf = self.arch_settings[model_cnf]
+        blocks = model_cnf[0]
+        self.n_blocks = sum(blocks)
+        channels = model_cnf[1]
+        reduce_ratios = [4, 2, 1, 1]
+        dpr = [x.item() for x in torch.linspace(0, drop_path, self.n_blocks)
+               ]  # stochastic depth decay rule
+        num_knn = [int(x.item()) for x in torch.linspace(k, k, self.n_blocks)
+                   ]  # number of knn's k
+        max_dilation = 49 // max(num_knn)
+
+        self.stem = Stem(out_dim=channels[0], act=act)
+        self.pos_embed = nn.Parameter(
+            torch.zeros(1, channels[0], 224 // 4, 224 // 4))
+        HW = 224 // 4 * 224 // 4
+
+        self.backbone = nn.ModuleList([])
+        idx = 0
+        for i in range(len(blocks)):
+            if i > 0:
+                self.backbone.append(Downsample(channels[i - 1], channels[i]))
+                HW = HW // 4
+            for j in range(blocks[i]):
+                self.backbone += [
+                    Seq(
+                        Grapher(
+                            channels[i],
+                            num_knn[idx],
+                            min(idx // 4 + 1, max_dilation),
+                            conv,
+                            act,
+                            norm,
+                            bias,
+                            use_stochastic,
+                            epsilon,
+                            reduce_ratios[i],
+                            n=HW,
+                            drop_path=dpr[idx],
+                            relative_pos=True),
+                        FFN(channels[i],
+                            channels[i] * 4,
+                            act=act,
+                            drop_path=dpr[idx]))
+                ]
+                idx += 1
+        self.backbone = Seq(*self.backbone)
+
+        self.prediction = Seq(
+            build_conv_layer(None, channels[-1], 1024, 1, bias=True),
+            nn.BatchNorm2d(1024), build_activation_layer(dict(type=act)),
+            nn.Dropout(dropout),
+            build_conv_layer(None, 1024, n_classes, 1, bias=True))
```

Comment:
The section should belong to the head, instead of the backbone.

Reply:
So I should put it in mmcls/models/heads, or use the script in mmcls/models/heads?